resources/sdk/purecloudjava/templates/ApiClient.mustache

```diff
@@ -975,4 +985,61 @@ public class ApiClient implements AutoCloseable {
         @Override
         public void close() throws Exception { }
     }
+
+     private ApiClientConnectorResponse retryInvoke(ApiClientConnectorRequest connectorRequest, ApiClientConnectorResponse connectorResponse) {
+         int retriesBeforeBackOff = 0;
+         long waitTimeInMillis;
+         int backOffRetries = 1;
+         long totalWaitTimeInMillis = 0L;
+         int MAX_RETRIES_BEFORE_BACKOFF = 5;
+         long FIVE_MINUTES_IN_MILLIS = 300000L;
+         long MAX_BACKOFF_IN_MILLIS = 27000L;
+         int MAX_BACKOFF_RETRIES = 3;
+
+         try {
+             do {
+                 if (connectorResponse.getHeaders().containsKey("Retry-After") && retriesBeforeBackOff++ < MAX_RETRIES_BEFORE_BACKOFF) {
+                     waitTimeInMillis = Integer.parseInt(connectorResponse.getHeaders().get("Retry-After")) * 1000L;
```

Comment:
Will this throw an exception if the retry after isn't set? That'd be possible if Public API is totally down

Reply:
yes, it may throw an exception during parsing and the original connectorResponse will be returned.